demisto_sdk/commands/common/hook_validations/playbook.py

```diff
@@ -85,6 +87,89 @@ def is_valid_version(self):  # type: () -> bool
         """
         return self._is_valid_version()
 
+    def collect_all_inputs_in_use(self):
+        """
+
+        Returns: list of all inputs used in playbook.
+
+        """
+        result: set = set()
+        with open(self.file_path, 'r') as f:
+            playbook_text = f.read()
+        all_inputs_occurrences = re.findall(r"inputs\.[-\w ]+", playbook_text)
+        for input in all_inputs_occurrences:
+            input = input.strip()
+            splitted = input.split('.')
+            if len(splitted) > 1:
+                result.add(splitted[1])
+        return result
+
+    def collect_all_inputs_from_inputs_section(self):
+        """
+
+        Returns: list of all inputs from 'inputs' section of playbook.
+
+        """
+        inputs: Dict = self.current_file.get('inputs', {})
+        inputs_keys = []
+        for input in inputs:
+            inputs_keys.append(input['key'])
+        return set(inputs_keys)
+
+    def inputs_in_use_check(self, is_modified: bool) -> bool:
+        """
+
+        Args:
+            is_modified: Wether the given files are modified or not.
+
+        Returns:
+            True if both directions for input use in playbook passes.
+
+        """
+        if not is_modified:
+            return True
+        inputs_in_use: set = self.collect_all_inputs_in_use()
+        inputs_in_section: set = self.collect_all_inputs_from_inputs_section()
+        return self.are_all_inputs_in_use(inputs_in_use, inputs_in_section) and \
+            self.are_all_used_inputs_in_inputs_section(inputs_in_use, inputs_in_section)
+
+    @error_codes('PB118')
+    def are_all_inputs_in_use(self, inputs_in_use: set, inputs_in_section: set) -> bool:
+        """Check whether the playbook inputs are in use in any of the tasks
+
+        Return:
+            bool. if the Playbook inputs are in use.
+        """
+
+        inputs_not_in_use = inputs_in_section.difference(inputs_in_use)
+
+        if inputs_not_in_use:
+            playbook_name = self.current_file.get('name', '')
+            error_message, error_code = Errors.input_key_not_in_tasks(playbook_name, ', '.join(inputs_not_in_use))
```

Comment:
Small _delight the customer_ add-on will be showing the inputs in alphabetical order (just call `sorted` before `join`ing) to make it easier to compare between runs, as order does not have other meanings here.